### PR TITLE
[code-infra] Fix failing Netlify build

### PR DESCRIPTION
**Pull request by cherniavskii**
After experimenting with different things in https://github.com/mui/mui-x/pull/13623 (you can see my commits if you're curious about what I tried), I managed to have the Netlify build working again.

I don't know what the origin of the issue is and why it's not reproducible locally after clearing the cache.

### Testing:
- I tested it in [`56f5131` (#13623)](https://github.com/mui/mui-x/pull/13623/commits/56f513194d3d615082628785ec13edfc3cc2c1e5) and it fixed the failing build.
- The ultimate test for this would be merging this PR and then merging `master` into one of the PRs that has the build failing

### Other attempts

- ❌ Make sure the correct version of `@mui/joy` installed on Netlify. [Commit](https://github.com/mui/mui-x/pull/13623/commits/ea42fe2dca7a447641d593726d597353d6dfc419) [Deployment](https://app.netlify.com/sites/material-ui-x/deploys/66ad352c40f2470008a9d38c)
- ❌ Remove Next.js build cache before build (`docs/.next`) [Commit](https://github.com/mui/mui-x/pull/13623/commits/33a5376c0a5262f7a87f50699b98e3abf9c0c2f6) [Deployment](https://app.netlify.com/sites/material-ui-x/deploys/66aec464dc07250008ec8c35)
- ❌ Merge the latest `master`. [Commit](https://github.com/mui/mui-x/pull/13623/commits/7d53c83edcbea624eee9c7ba023a0395a2acaabe)  [Deployment](https://app.netlify.com/sites/material-ui-x/deploys/66b3845c61be71000859cae4)
- ❌ Bump a dependency to update the lockfile and invalidate the old cache [Commit](https://github.com/mui/mui-x/pull/13623/commits/106cc31ad09f089c428ce0cb4693e229b5c63fb7) [Deployment](https://app.netlify.com/sites/material-ui-x/deploys/66b4d97209a47200083aac11)
- ❌ Downgrade pnpm version.
  - 9.4.0 [Commit](https://github.com/mui/mui-x/pull/13623/commits/d3aff1415c1be2031a27a6ed7513869bf5e1407e) [Deployment](https://app.netlify.com/sites/material-ui-x/deploys/66ba327c22e08c000824d00f)
  - 9.3.0 [Commit](https://github.com/mui/mui-x/pull/13623/commits/dff87504fa7de5c1d2c9d474dadaef598d15fbe1) [Deployment](https://app.netlify.com/sites/material-ui-x/deploys/66ba35ab1c659700088c3c98)

### CODE REVIEW

**Comment by JCQuintas** [comment]
Did you have to do any change on `@mui/monorepo/docs/pages/_document` or it was simply copy/paste?

File: docs/pages/_document.js
Comment (cherniavskii) [inline docs/pages/_document.js:1]: @JCQuintas It's pure copy/paste from the latest `next` branch of the monorepo, not a single character changed:
https://github.com/mui/material-ui/blob/f9de2dd508b59623a99949f8891b96cf0016581c/docs/pages/_document.js

**Comment by Janpot** [comment]
Hmm, the plan with `@mui/docs` was to centralize _app.js and _document.js content in there, not duplicate more 😬 

> I don't know what the origin of the issue is and why it's not reproducible locally after clearing the cache.

🤔 Now that I come to think of it, I feel like lately I've been seeing monorepo changes come in to Toolpad without merging a version bump PR. If suddenly monorepo changes come in without

**Comment by LukasTy** [comment]
It looks like our Netlify config might not be ideal based on [their doc](https://docs.netlify.com/configure-builds/monorepos/) and the following points [from](https://answers.netlify.com/t/support-guide-my-site-deploy-fails-unless-netlifys-build-cache-is-cleared/3035).
> 5. If you are using [yarn workspaces 14](https://docs.netlify.com/configure-builds/monorepos/#yarn-workspaces) or a [monorepo 16](https://docs.netlify.com/configure-builds/monorepos/#build-a-single-site-from-a-subdirectory-or-monorepo) please consult the appropriate documentation (linked to in this sentence).
> 6. Fairly rare but worth mentioning: You have a package.json in your repo root, but you are building a subdirectory of your repo and have NOT appropriately set the base directory in your build settings. That can lead to some confusing caching situations since we’ll only sometimes use that package.json (when there is nocache).

WDYT about trying this approach? 🤔 
https://github.com/mui/mui-x/pull/14105

**Pull request closed without merging**